### PR TITLE
Cherry pick for 0.56 release

### PR DESCRIPTION
**Pull request by sklam**
as titled.

### CODE REVIEW

File: CHANGE_LOG [hunk 1]
```diff
@@ -265,12 +265,16 @@ Pull-Requests:
...
+* PR `#8192 <https://github.com/numba/numba/pull/8192>`_: Update CHANGE_LOG for 0.56.0 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
 * PR `#8195 <https://github.com/numba/numba/pull/8195>`_: Make the workqueue threading backend once again fork safe. (`stuartarchibald <https://github.com/stuartarchibald>`_)
 * PR `#8196 <https://github.com/numba/numba/pull/8196>`_: Fix numerical tolerance in parfors caching test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
 * PR `#8197 <https://github.com/numba/numba/pull/8197>`_: Fix `isinstance` warning check test. (`stuartarchibald <https://github.com/stuartarchibald>`_)
 * PR `#8203 <https://github.com/numba/numba/pull/8203>`_: pin llvmlite 0.39 for public CI builds (`esc <https://github.com/esc>`_)
 * PR `#8192 <https://github.com/numba/numba/pull/8192>`_:  Update CHANGE_LOG for 0.56.0 (`stuartarchibald <https://github.com/stuartarchibald>`_ `esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:273]: I think this line should be removed, as it is out of order, incorrectly indented, and covered above by a new correct entry in the right place.

File: CHANGE_LOG [hunk 2]
```diff
@@ -265,12 +265,16 @@ Pull-Requests:
...
 * PR `#8205 <https://github.com/numba/numba/pull/8205>`_:  Pin llvmlite and NumPy on release branch.(`esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:274]:
```suggestion
* PR `#8205 <https://github.com/numba/numba/pull/8205>`_: Pin llvmlite and NumPy on release branch.(`esc <https://github.com/esc>`_ `Siu Kwan Lam <https://github.com/sklam>`_)
```

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch. I've checked this manually by replaying the cherry-picks of the merges and resolving conflicts similarly and it seems to match up!

**Comment by sklam** [comment]
enough CI has passed for doc changes. going to merge

**Pull request merged**